prover/src/matrix/tests.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use crate::{
+    math::{fields::f64::BaseElement, get_power_series, log2, polynom, StarkField},
+    Matrix, RowMatrix,
+};
+use math::FieldElement;
+use rand_utils::rand_vector;
+use utils::collections::Vec;
+
+#[test]
+fn test_eval_poly_with_offset_matrix() {
+    let n = 256;
+    let num_polys = 64;
+    let blowup_factor = 8;
+
+    // generate random columns. Each column is a polynomial of degree n - 1.
+    let mut columns: Vec<Vec<BaseElement>> = (0..num_polys).map(|_| rand_vector(n)).collect();
+
+    // evaluate columns using the row matrix implementation.
+    let row_matrix = RowMatrix::from_polys(&Matrix::new(columns.clone()), blowup_factor);
+
+    // evaluate columns using the using the polynomial evaluation implementation.
+    let offset = BaseElement::GENERATOR;
+    let domain = build_domain(n * blowup_factor);
+    let shifted_domain = domain.iter().map(|&x| x * offset).collect::<Vec<_>>();
+    for p in columns.iter_mut() {
+        *p = polynom::eval_many(p, &shifted_domain);
+    }
+
+    // transpose the columns back to a row major format.
+    let eval_col = transpose(columns);
```

Comment:
Instead of doing a transpose, could we compare `row_matrix` and `columns` row-by-row? We can just implement a simple `get_row()` function and use it in a loop. The function could look like this:
```Rust
fn get_row(columns: &Vec<Vec<BaseElement>>, row_id: usize) -> Vec<BaseElement>
```